sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -83,17 +88,27 @@ def _parse_conn_str(conn_str):
             ):  # Fallback since technically expiry is optional.
                 # An arbitrary, absurdly large number, since you can't renew.
                 shared_access_signature_expiry = int(time.time() * 2)
-    if not (
-        all((endpoint, shared_access_key_name, shared_access_key))
-        or all((endpoint, shared_access_signature))
-    ) or all(
-        (shared_access_key_name, shared_access_signature)
-    ):  # this latter clause since we don't accept both
+
+    if not endpoint:
+        raise ValueError("Connection string is either blank or malformed.")
+    parsed = urlparse(endpoint.rstrip("/"))
+    if not parsed.netloc:
+        raise ValueError("Invalid Endpoint on the Connection String.")
```

Comment:
I think we're checking the fully qualified namespace here, wondering if there's any change we could combine it with `host = cast(str, strip_protocol_from_uri(cast(str, endpoint)))`

Reply:
since `parsed.netloc` already strips the protocol, I think that can be substituted for `strip_protocol_from_uri(cast(str, endpoint))`